platform/mellanox/mlnx-platform-api/sonic_platform/fan.py

```diff
@@ -48,6 +55,45 @@ def __init__(self, fan_index, drawer_index = 1, psu_fan = False):
         self.fan_orange_led_path = "led_fan{}_orange".format(self.drawer_index)
         self.fan_pwm_path = "pwm1"
         self.fan_led_cap_path = "led_fan{}_capability".format(self.drawer_index)
+        if sku in skus_with_fan_dir:
+            self.fan_dir = FAN_DIR
+        else:
+            self.fan_dir = None
+
+
+    def get_direction(self):
+        """
+        Retrieves the direction of fan
+
+        Returns:
+            A string, either FAN_DIRECTION_INTAKE or FAN_DIRECTION_EXHAUST
+            depending on fan direction
+
+        Notes:
+            What Mellanox calls forward: 
+            Air flows from fans side to QSFP side, for example: MSN2700-CS2F
+            which means intake in community
+            What Mellanox calls reverse:
+            Air flow from QSFP side to fans side, for example: MSN2700-CS2R
+            which means exhaust in community
+            According to hw-mgmt:
+                1 stands for forward, in other words intake
+                0 stands for reverse, in other words exhaust
+        """
+        if not self.fan_dir or self.is_psu_fan:
+            raise NotImplementedError
```

Comment:
my view is that in this case, it's better to set the fan direction to 'unknown', maybe can return None?

Reply:
not sure whether it is a better way and to do so requires to update platform-common.
will discuss with Joe and find a solution.